### PR TITLE
Unify the runtime folder for ehsm

### PR DESCRIPTION
**Pull request by wanghouqi**
Change the running folder of dkeyserver and dkeycache to /var/run/ehsm/

test: passed the k8s cluster

Signed-off-by: wanghouqi <houqix.wang@intel.com>

### CODE REVIEW

File: dkeycache/App/la_server.h [hunk 1]
```diff
@@ -34,6 +34,9 @@
 
 #include "la_task.h"
 
+#define RUNTIME_FOLDER "/var/run/ehsm/" // the folder is used to store the files generated by the runtime
+#define UNIX_DOMAIN "/var/run/ehsm/dkeyprovision.sock"
+
```
Comment (syan10) [inline dkeycache/App/la_server.h:39]: 建议把/var/run/ehsm/ 这个定义 buildenv.mk里面

File: dkeycache/App/main.cpp [hunk 2]
```diff
@@ -162,6 +163,15 @@ int main(int argc, char *argv[])
     // process argv
     parse_args(argc, argv);
 
+    // mkdir RUNTIME_FOLDER
+    if (access(RUNTIME_FOLDER, F_OK) != 0) {
+        log_i("Initializing runtime folder [path: %s].", RUNTIME_FOLDER);
+        if (mkdir(RUNTIME_FOLDER, 0755) != 0) {
+            log_e("Create runtime folder failed!");
```
Comment (yang8621) [inline dkeycache/App/main.cpp:170]: return?

File: dkeycache/App/la_server.h (hunk 1 above)
Comment (wanghouqi) [reply dkeycache/App/la_server.h:40]: fixed

File: dkeycache/App/main.cpp (hunk 2 above)
Comment (wanghouqi) [reply dkeycache/App/main.cpp:170]: fixed

File: buildenv.mk [hunk 3]
```diff
@@ -39,6 +39,10 @@ VERSION_STRING := $(vsn_major).$(vsn_minor).$(vsn_patch)
 DATE_STRING := `date "+20%y.%m.%d %k:%M"`
 EHSM_KMS_GIT_SHA=$(shell git rev-parse --short=7 --verify HEAD)
```
Comment (syan10) [inline buildenv.mk:41]: title:
"
Unify the runtime folder for ehsm
"

Comment (wanghouqi) [reply buildenv.mk:41]: fixed

File: core/Makefile [hunk 4]
```diff
@@ -184,6 +184,7 @@ App/auto_version.h:
 	@echo "#define EHSM_VERSION \"$(VERSION_STRING)\"" >> App/auto_version.h
 	@echo "#define EHSM_DATE \"$(DATE_STRING)\""	>> App/auto_version.h
 	@echo "#define EHSM_GIT_SHA \"$(EHSM_KMS_GIT_SHA)\""	>> App/auto_version.h
+	@echo "#define RUNTIME_FOLDER \"$(RUNTIME_FOLDER)\""	>> App/auto_version.h
```
Comment (yang8621) [inline core/Makefile:187]: 放这合适吗： auto_version.h

Comment (yang8621) [reply core/Makefile:187]: 还是参考64行用 -D 吧
-DSIGNED_ENCLAVE_FILENAME=\"$(Signed_Enclave_FileName)\" -DSGX_SIGNING_TOOL=\"$(Sgx_Signing_tool)\"

File: buildenv.mk [hunk 5]
```diff
@@ -39,6 +39,10 @@ VERSION_STRING := $(vsn_major).$(vsn_minor).$(vsn_patch)
...
 
+######## ehsm run folder  ########
+# the folder is used to store the files generated by the runtime
```
Comment (yang8621) [inline buildenv.mk:43]: 加一句注释:
Pease always keep the last /slash/

Comment (wanghouqi) [reply buildenv.mk:43]: fixed

File: core/Makefile (hunk 4 above)
Comment (wanghouqi) [reply core/Makefile]: fixed

File: core/App/fifo.cpp [hunk 6]
```diff
@@ -42,13 +43,14 @@
 
 #include "fifo_def.h"
 #include "log_utils.h"
+#include "auto_version.h"
```
Comment (yang8621) [inline core/App/fifo.cpp:46]: remove？

File: dkeycache/App/la_server.h [hunk 7]
```diff
@@ -33,6 +33,10 @@
 #define _LA_SERVER_H_
 
 #include "la_task.h"
+#include "auto_version.h"
```
Comment (yang8621) [inline dkeycache/App/la_server.h:36]: remove？

Comment (wanghouqi) [reply dkeycache/App/la_server.h]: fixed

File: core/App/fifo.cpp (hunk 6 above)
Comment (wanghouqi) [reply core/App/fifo.cpp]: fixed

**Review by yang8621 — APPROVED** [review]

**Pull request merged**